chains/evm/contracts/pools/USDC/CCTPV2/USDCTokenPoolCCTPV2.sol

```diff
@@ -0,0 +1,163 @@
+  /// The allowedCaller is preconfigured per destination domain and token pool version refer Domain struct.
+  /// @dev Emits ITokenMessenger.DepositForBurn event.
+  /// @dev Assumes caller has validated the destinationReceiver.
+  function lockOrBurn(
+    Pool.LockOrBurnInV1 calldata lockOrBurnIn
+  ) public virtual override returns (Pool.LockOrBurnOutV1 memory) {
+    _validateLockOrBurn(lockOrBurnIn);
+
+    USDCTokenPool.Domain storage domain = s_chainToDomain[lockOrBurnIn.remoteChainSelector];
+
+    if (!domain.enabled) revert UnknownDomain(lockOrBurnIn.remoteChainSelector);
+
+    if (lockOrBurnIn.receiver.length != 32) {
+      revert InvalidReceiver(lockOrBurnIn.receiver);
+    }
+
+    // To support certain non-EVM chains, the mint recipient may be overridden to be a token pool which then
+    // forwards the tokens to the receiver. The message itself will not be changed and the destination token pool will
+    // still receive the correct address of the final token receiver.
+    bytes32 decodedReceiver;
+    if (domain.mintRecipient != bytes32(0)) {
+      decodedReceiver = domain.mintRecipient;
+    } else {
+      decodedReceiver = abi.decode(lockOrBurnIn.receiver, (bytes32));
+    }
+
+    // Since this pool is the msg sender of the CCTP transaction, only this contract
+    // is able to call replaceDepositForBurn. Since this contract does not implement
+    // replaceDepositForBurn, the tokens cannot be maliciously re-routed to another address.
+
+    // Since the CCTP message will use slow-burn, the maxFee is 0, and the finality threshold is standard (2000).
+    // Using fast-burn would require a maxFee and a finality threshold of 1000, which may be added in the future.
+
+    // In CCTP V2, nonces are deterministic and not sequential. As a result the nonce is not returned to this contract
+    // upon sending the message, and will therefore not be included in the destPoolData. It will instead be
+    // acquired off-chain and included in the destination-message's offchainTokenData.
+    i_tokenMessenger.depositForBurn(
+      lockOrBurnIn.amount, // amount
+      domain.domainIdentifier, // destinationDomain
+      decodedReceiver, // mintRecipient
+      address(i_token), // burnToken
+      domain.allowedCaller, // destinationCaller
+      CCTPV2.MAX_FEE, // maxFee
+      CCTPV2.FINALITY_THRESHOLD // minFinalityThreshold
+    );
+
+    emit LockedOrBurned({
+      remoteChainSelector: lockOrBurnIn.remoteChainSelector,
+      token: address(i_token),
+      sender: msg.sender,
+      amount: lockOrBurnIn.amount
+    });
+
+    // Since CCTP V2 does not return a nonce during the deposit call, we can just use zero to satisfy the struct field.
+    return Pool.LockOrBurnOutV1({
+      destTokenAddress: getRemoteToken(lockOrBurnIn.remoteChainSelector),
+      destPoolData: abi.encode(
+        SourceTokenDataPayload({nonce: 0, sourceDomain: i_localDomainIdentifier, cctpVersion: CCTPVersion.VERSION_2})
```

Comment:
I'd like to add the `depositForBurn` params to `SourceTokenDataPayload`, e.g. `destinationDomain`, `decodedReceiver` (i.e. `mintRecipient`), etc